src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js

```diff
@@ -1,44 +1,173 @@
-import React from "react";
-import { Link } from "gatsby";
+import React, { useState, useEffect } from "react";
+import { Link, useStaticQuery, graphql } from "gatsby";
 import { HoneycombGrid } from "./Integration.style";
+import { ResponsiveHoneycomb, Hexagon } from "react-honeycomb";
+
+const IntegrationsGrid = ({ category }) => {
+
+  const data = useStaticQuery(graphql`
+  query {
+    allMdx(filter: { fields: { collection: { eq: "integrations" } } }) {
+      nodes {
+        frontmatter {
+          title
+          status
+          category
+          integrationIcon {
+            childImageSharp {
+              gatsbyImageData(width: 500, layout: CONSTRAINED)
+            }
+            extension
+            publicURL
+          }
+        }
+        fields{
+          slug
+        }
+      }
+    }
+  }
+   `);
+
+  const [IntegrationList, setIntegrationList] = useState(data.allMdx.nodes);
+
+  let [categoryList,setCategoryList] = useState([
+    { id: 1, name: "All", isSelected: false },
+    { id: 2, name: "Platforms", isSelected: false },
+    { id: 3, name: "Service Mesh", isSelected: false },
+    { id: 4, name: "Operating System", isSelected: false },
+    { id: 5, name: "Collaboration", isSelected: false },
+  ]);
+
+  useEffect(() => setCategory(), []);
+
+  const setCategory = () => {
+    if (category !== undefined){
+      categoryList.forEach((item) => {
+        if (item.name === category) item.isSelected = true;
+      });
+    } else {
+      categoryList[0].isSelected = true;
+    }
+
+    setCategoryList(categoryList);
+    setIntegrationCollection();
+
```

Comment:
Remove such extra line spacings from the code

Reply:
I added extra line spacing to make code more readable. :)

Reply:
That's fine, but in most places, it is redundant. Like no extra line is needed after a block code ends. Remove them

Reply:
Okay!